templates/boltzmann/core/prelude.ts

```diff
@@ -47,12 +53,74 @@ if (!process.env.HONEYCOMB_TEAM && process.env.HONEYCOMBIO_TEAM) {
   process.env.HONEYCOMB_TEAM = process.env.HONEYCOMBIO_TEAM
 }
 
-beeline({
-  writeKey: process.env.HONEYCOMB_WRITEKEY,
-  dataset: process.env.HONEYCOMB_DATASET,
-  sampleRate: Number(process.env.HONEYCOMB_SAMPLE_RATE) || 1,
-  serviceName,
-})
+if (!process.env.HONEYCOMB_DATASET && process.env.HONEYCOMBIO_DATASET) {
+  process.env.HONEYCOMB_DATASET = process.env.HONEYCOMBIO_DATASET
+}
+
+let sdk: NodeSDK | null = null
+
+if (
+  process.env.HONEYCOMB_WRITE_KEY &&
+  process.env.HONEYCOMB_DATASET &&
+  process.env.HONEYCOMB_API_HOST
+) {
+  const metadata = new Metadata()
+  metadata.set('x-honeycomb-team', process.env.HONEYCOMB_WRITE_KEY)
+  metadata.set('x-honeycomb-dataset', process.env.HONEYCOMB_DATASET)
+
+  const traceExporter = new CollectorTraceExporter({
+    url: process.env.HONEYCOMB_API_HOST,
+    credentials: credentials.createSsl(),
+    metadata
+  })
+
+  sdk = new NodeSDK({
+    resource: new Resource({
+      [SemanticResourceAttributes.SERVICE_NAME]: serviceName
+    }),
+    traceExporter,
+    instrumentations: [getNodeAutoInstrumentations()],
+    metricInterval: Math.floor(
+      1 / (Number(process.env.HONEYCOMB_SAMPLE_RATE) || 1)
+    )
+  })
+} else {
+  beeline({
+    writeKey: process.env.HONEYCOMB_WRITEKEY,
+    dataset: process.env.HONEYCOMB_DATASET,
+    sampleRate: Number(process.env.HONEYCOMB_SAMPLE_RATE) || 1,
+    serviceName,
+  })
+}
+
+function initOtelSDK(): Promise<void> {
+  if (sdk) {
+    let exitCode = 0
+    process.once('SIGTERM', shutdown)
+    process.once('beforeExit', shutdown)
+    process.once('uncaughtException', die)
+    process.once('unhandledRejection', die)
+    return sdk.start()
+
+    async function die(err: Error) {
+      console.error(err.stack)
+      exitCode = 1
+      await shutdown()
+    }
+
+    async function shutdown() {
+      if (sdk) {
+        try {
+          await sdk.shutdown()
+        } catch (err) {
+          console.error(err.stack)
+        }
+      }
+      process.exit(exitCode)
```

Comment:
The CI repo has a "superstitious" timeout in it. I didn't replicate that here, but can if people think it's appropriate.